python/ray/data/_internal/datasource/parquet_datasink.py

```diff
@@ -109,70 +137,85 @@ def write_blocks_to_path():
             max_backoff_s=WRITE_FILE_RETRY_MAX_BACKOFF_SECONDS,
         )
 
-    def _write_single_file(
+    def _get_basename_template(self, filename: str, write_uuid: str) -> str:
+        # Check if write_uuid is present in filename, add if missing
+        if write_uuid not in filename and self.mode == SaveMode.APPEND:
+            raise ValueError(
+                f"Write UUID '{write_uuid}' not found in filename '{filename}'. "
+                f"Please modify your FileNameProvider implementation to include the write_uuid in the filename. "
+                f"Adding UUID as suffix to ensure uniqueness across write operations."
```

Comment:
```suggestion
                f"Write UUID '{write_uuid}' missing from filename template '{filename}'. This could result in files being overwritten."
                f"Modify your FileNameProvider implementation to include the `write_uuid` into the filename template or change your write mode to SaveMode.OVERWRITE. "
```